src/connectors/snowflake/trulens/connectors/snowflake/utils/server_side_evaluation_artifacts.py

```diff
@@ -0,0 +1,221 @@
+                os.path.join(data_directory, zip_to_upload),
+                f"@{_STAGE_NAME}",
+            )
+
+    def _set_up_stream(self) -> None:
+        self._run_query(
+            f"""
+            CREATE STREAM IF NOT EXISTS {_STREAM_NAME}
+                ON TABLE {self._database_name}.{self._schema_name}.TRULENS_FEEDBACKS
+                SHOW_INITIAL_ROWS = TRUE
+            """
+        )
+
+    def _set_up_stored_procedure(self) -> None:
+        imports = ",".join([
+            f"'@{_STAGE_NAME}/{curr}'" for curr in _ZIPS_TO_UPLOAD
+        ])
+        self._run_query(
+            f"""
+            CREATE PROCEDURE IF NOT EXISTS {_STORED_PROCEDURE_NAME}()
+                RETURNS STRING
+                LANGUAGE PYTHON
+                RUNTIME_VERSION = '3.11'
+                PACKAGES = (
+                    -- TODO(dkurokawa): get these package versions automatically.
+                    'alembic',
+                    'dill',
+                    'munch',
+                    'nest-asyncio',
+                    'nltk',
+                    'numpy',
+                    'packaging',
+                    'pandas',
+                    'pip',
+                    'pydantic',
+                    'python-dotenv',
+                    'requests',
+                    'rich',
+                    'scikit-learn',
+                    'scipy',
+                    'snowflake-snowpark-python',
+                    'sqlalchemy',
+                    'tqdm',
+                    'typing_extensions'
+                )
+                IMPORTS = ({imports})
+                HANDLER = 'run'
+            AS
+                $$
+            import _snowflake
+            import trulens.providers.cortex.provider
+            from snowflake.sqlalchemy import URL
+            from trulens.core import TruSession
+            from trulens.core.schema.feedback import FeedbackRunLocation
+
+
+            def run(session):
+                # Set up sqlalchemy engine parameters.
+                conn = session.connection
+                engine_params = {{}}
+                engine_params["paramstyle"] = "qmark"
+                engine_params["creator"] = lambda: conn
+                database_args = {{"engine_params": engine_params}}
+                # Ensure any Cortex provider uses the only Snowflake connection allowed in this stored procedure.
+                trulens.providers.cortex.provider._SNOWFLAKE_STORED_PROCEDURE_CONNECTION = (
+                    conn
+                )
+                # Run deferred feedback evaluator.
+                db_url = URL(
+                    account="{self._account}",
+                    user="{self._user}",
+                    password="password",
+                    database="{self._database_name}",
+                    schema="{self._schema_name}",
+                    warehouse="{self._warehouse}",
+                    role="{self._role}",
+                )
```

Comment:
no you need it a bit. It's what decides to even use a snowflake dialect at all. AFAICT you don't need the password though since the connection is used.